test/flops.h

```diff
@@ -112,18 +112,18 @@ static double  flops_sgemm(double m, double n, double k)
 static double fmuls_gbmm(double m, double n, double k, double kl, double ku)
 { 
     if(m > k+kl)
-        return (kl*k + (k-1.0)*k*0.5 - (k-ku-1.0)*(k-ku)*0.5)*n;
+        return (kl*k + (k+1.0)*k*0.5 - (k-ku-1.0)*(k-ku)*0.5)*n;
     if(k > m+ku)
-        return (ku*m - (m-kl-1.0)*(m-kl)*0.5 + (m-1.0)*m*0.5)*n;
+        return (ku*m - (m-kl-1.0)*(m-kl)*0.5 + (m+1.0)*m*0.5)*n;
     return (m*k - (m-kl-1.0)*(m-kl)*0.5 - (k-ku-1.0)*(k-ku)*0.5)*n;
 }
 
 static double fadds_gbmm(double m, double n, double k, double kl, double ku)
 { 
     if(m > k+kl)
-        return (kl*k + (k-1.0)*k*0.5 - (k-ku-1.0)*(k-ku)*0.5)*n;
+        return (kl*k + (k+1.0)*k*0.5 - (k-ku-1.0)*(k-ku)*0.5)*n;
     if(k > m+ku)
-        return (ku*m - (m-kl-1.0)*(m-kl)*0.5 + (m-1.0)*m*0.5)*n;
+        return (ku*m - (m-kl-1.0)*(m-kl)*0.5 + (m+1.0)*m*0.5)*n;
```

Comment:
Formulas with `-` are more likely to be correct because they nicely deteriorate to `0` for special bandwidth sizes. This new formula loses this property.

Reply:
Forgive me for failing to understand, but I don't see how the bandwidth size has anything to do with the `-`. This change was due to a failure to properly simplify the algebra on my first formula, which I fixed here. Whether the `-` was a `+` will have no effect on simplifying for specific bandwidths, for it is about whether or not `1` will be added or subtracted from `m` or `k`, neither of which relates to the bandwidth.

I will push the first change regarding the space and wait for your feedback. Could you help point towards what I'm missing?

Reply:
I was simply talking about the fact that `m-1` becomes `0` when `m` is `1`. This is the correct result since there is nothing to do when `m==1`. There seems to be little foresight or safeguards in the initial formula and its fix. Just patching the patches and discussing the discussion. I asked for corner cases that could server as easy and verifiable assertions: with the triple of `m,n,k==7,5,3` there is a manageable small set of `kl,ku` pairs to verify. The general guideline is that the code that goes into the mathematical library is a fraction of the code used for testing and is left out. PR discussion is best left for design, form, or cross-cutting issues but not for correctness.

Reply:
In the third expression which you are commenting about, in the corner case when `m==1`: there *is* work to be done! The first expression accounts for the terms in the band, the second expression (which, like I said, was incorrect the first time) needs to account for the term in the diagonal. It would draw a triangle for a `m>1` case, but for `m=1` this is only a single element. The third term (which would subtract a triangle) *will* have this property of nicely deteriorating to zero. Just like the mainline equation case in LAWN41, my equation case will have exactly one of its terms nicely deteriorate to zero. I've attached a picture to explain my point. [The case I was talking about.](https://docs.google.com/presentation/d/1b8DSjmpInchSwy9-eGnyXlhHgkhL2TCVAp8jmQRwupw/edit?usp=sharing)

![LAWN41-GBMM-explain-m-1](https://user-images.githubusercontent.com/36454249/145848977-fdf3cbbe-fc3e-4e71-a8c4-dac403de7ea3.png)


What I *do* agree with, however, is that the form of this equation could perhaps be improved. Look at that rectangle in the equation. Notice that it happens to always be able to be one element larger. It could absorb this element, make the triangle smaller, and then reclaim the property of nicely deteriorating to zero. Of course, in this case, we would also lose out on a deterioration for `kl=0`. But why stop there: we could just combined the first two terms into the calculation of a trapezoid. I do trust your judgement on this call, as your intuition on what properties equations should have is surely more developed than mine ([m=3 case](https://docs.google.com/presentation/d/1b8DSjmpInchSwy9-eGnyXlhHgkhL2TCVAp8jmQRwupw/edit?usp=sharing)).

![LAWN41-GBMM-explain-m-3](https://user-images.githubusercontent.com/36454249/145849027-65e2cefd-9567-485f-be06-fbcd5736f95f.png)


I do agree that test cases would be helpful for review, so I went ahead and performed several and attached them to this thread so you can review them if it's helpful. As for discussing correctness in a PR, I'm conflicted. I would like to point out that you also committed an equation that produced incorrect results for flops (to be precise, this equation was added to PLASMA as one of my PRs was closed, a PR whose code *did* produce the correct answer for flops for rectangular matrices). Furthermore, the author of the equation in the BLAS apparently did not test for the case of a rectangular band matrix either. If discussing correctness here is not appropriate, I will happily comply with another method you suggest.

In the general case, however, I agree and in the future will be sure that questions on the correctness of my code are answered well before the PR stage begins.

Finally, forgive me for asking for clarification, but what does "The general guideline is that the code that goes into the mathematical library is a fraction of the code used for testing and is left out" mean? We leave out the code that goes into the mathematical library? This does not seem right.

simple test cases: [gbmm_flops_tester.zip](https://github.com/icl-utk-edu/plasma/files/7705247/gbmm_flops_tester.zip)